### PR TITLE
[MRG] lag features

### PR DESCRIPTION
**Pull request by solegalli**
Hi @Morgan-Sell 

I think we are done with this class. I modified the code a bit here and there. Re-organized the tests. Added some new tests. And also finished the documentation. I also removed some files that we had duplicated.

I re-organized the files of docs for other transformers, because now we have so many that the index was messy. You don't need to look at those files. I tagged your name in the file that I would appreciate if you could read, and let me know if it is clear and also fix any typo or grammar mistake that I may have made.

If you could have a look at the final state of the LagFeatures class and tests, and the doc file that I tagged you with below, and are happy with all those changes, could you please merge, and then in the main repo ask @KishManani for a final review?

PS: I know you are busy, so if you don't have time, don't worry, please merge and we take it from there  :)

### CODE REVIEW

File: docs/user_guide/timeseries/forecasting/LagFeatures.rst [hunk 1]
```diff
@@ -0,0 +1,257 @@
+.. _lag_features:
+
```
Comment (solegalli) [inline docs/user_guide/timeseries/forecasting/LagFeatures.rst:2]: @Morgan-Sell could you have a look at this doc file?

File: docs/user_guide/timeseries/forecasting/LagFeatures.rst [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+.. currentmodule:: feature_engine.timeseries.forecasting
+
+LagFeatures
+===========
+
+The :class:`LagFeatures` adds lag features to the dataframe. A lag feature is a feature
+with information about a prior time step.
+
+In forecasting, past values of the variable we want to forecast, are likely to be
+predictive. Past values of other accompanying predictive features can also help us
```
Comment (Morgan-Sell) [inline docs/user_guide/timeseries/forecasting/LagFeatures.rst:12]: I would probably write this sentence.

In forecasting, the historic data of both the predictive and accompanying features are used. Therefore, it is common practice to create lag features for time-series and predictive variables.

File: docs/user_guide/timeseries/forecasting/LagFeatures.rst (hunk 1 above)
Comment (Morgan-Sell) [reply docs/user_guide/timeseries/forecasting/LagFeatures.rst:2]: Looks good! I had one comment to make the text more concise.

**Comment by Morgan-Sell** [comment]
Hi @solegalli,

I left one minor comment. I believe the revision simplifies the text. Let me know if you would like to change the text. Otherwise, I can merge the PR.

File: docs/user_guide/timeseries/forecasting/LagFeatures.rst (hunk 2 above)
Comment (solegalli) [reply docs/user_guide/timeseries/forecasting/LagFeatures.rst:12]: the thing is, for this sentence, I borrowed Kishan's words. So let's just merge and then, if he finds it funny he can change them :p

**Comment by solegalli** [comment]
Thank you @Morgan-Sell 

If you could merge, I will discuss the sentence change with Kishan in the main repo. I borrowed his words, i think it is forecasting jargon. Thanks a lot!!!

**Pull request merged**